packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -905,7 +906,15 @@ function collectParams(
       if (value.type.kind === "Model") {
         collectParams(value.type.properties, params);
       } else {
-        params.push(value);
+        let sourceProp = value;
+        while (sourceProp.sourceProperty) {
+          sourceProp = sourceProp.sourceProperty;
+        }
+        if (sourceProp.model && !isAzureCoreModel(sourceProp.model)) {
```

Comment:
for templated type in core lib like [`Page`](https://github.com/Azure/typespec-azure/blob/955fc9534c2f7153d45c9fbe5743baa0c104d803/packages/typespec-azure-core/lib/models.tsp#L17), it will also be a core model judged by `isAzureCoreModel`. i don't know if it will happen for param, but i think we should fix the `isAzureCoreModel`.
```
@useDependency(Versions.v1_0_Preview_2)
@server("http://localhost:3000", "endpoint")
@service()
namespace My.Service;
@resource("widgets")
model Widget {
  @key("widgetName")
  @visibility("read")
  name: string;
  manufacturerId: string;
}

op listWidgets is ResourceList<Widget>;
```

Reply:
i'm not able to repro this issue with the test you linked. Since I still don't fully understand the issue and how to repro it, I'm going to go ahead and merge this pr. Do you have another repro link? I can take a look at this tomorrow. Thanks @tadelesh!

Reply:
it's fine. since your purpose is just to ignore the parameter from core model spread, the logic should be good. what i'm saying is for `isAzureCoreModel`, sth like `Azure.Core.Page<Widget>` will also be considered as a core model which should be wrong.

Reply:
I still can't repro it. I've added a test to the pr and it's still just showing a single model, feel free to add to it after I merge this PR